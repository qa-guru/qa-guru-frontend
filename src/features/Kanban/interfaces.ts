export interface ICard {
  id: string;
  title: string;
  description: string;
}

export interface IColumn {
  id: string;
  title: string;
  cards: ICard[];
}
