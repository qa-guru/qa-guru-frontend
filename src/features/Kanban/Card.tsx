import React from "react";
import { useDrag } from "react-dnd";
import { ICard } from "./interfaces";

interface CardProps {
  card: ICard;
  sourceColumnId: string;
}

const Card: React.FC<CardProps> = ({ card, sourceColumnId }) => {
  const [{ isDragging }, dragRef] = useDrag({
    type: "card",
    item: { id: card.id, sourceColumnId },
    collect: (monitor) => ({
      isDragging: monitor.isDragging(),
    }),
  });

  const opacity = isDragging ? 0.5 : 1;

  return (
    <div
      ref={dragRef}
      style={{
        backgroundColor: "white",
        padding: "8px",
        borderRadius: "4px",
        marginBottom: "8px",
        boxShadow: "0 1px 4px rgba(0, 0, 0, 0.2)",
        opacity,
      }}
    >
      <h4>{card.title}</h4>
      <p>{card.description}</p>
    </div>
  );
};

export default Card;
