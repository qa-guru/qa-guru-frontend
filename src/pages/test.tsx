import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
