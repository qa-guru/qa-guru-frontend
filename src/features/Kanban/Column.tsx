import React from "react";
import { useDrop } from "react-dnd";
import Card from "./Card";
import { IColumn, ICard } from "./interfaces";

interface ColumnProps {
  column: IColumn;
  onCardDrop: (
    cardId: string,
    sourceColumnId: string,
    targetColumnId: string
  ) => void;
}

const Column: React.FC<ColumnProps> = ({ column, onCardDrop }) => {
  const [{ isOver }, dropRef] = useDrop({
    accept: "card",
    drop: (item: { id: string; sourceColumnId: string }) => {
      onCardDrop(item.id, item.sourceColumnId, column.id);
    },
    collect: (monitor) => ({
      isOver: monitor.isOver(),
    }),
  });

  const backgroundColor = isOver ? "#e0e0e0" : "#f0f0f0";

  return (
    <div
      ref={dropRef}
      style={{
        backgroundColor,
        padding: "8px",
        borderRadius: "4px",
        width: "250px",
      }}
    >
      <h3>{column.title}</h3>
      {column.cards.map((card) => (
        <Card key={card.id} card={card} sourceColumnId={column.id} />
      ))}
    </div>
  );
};

export default Column;
