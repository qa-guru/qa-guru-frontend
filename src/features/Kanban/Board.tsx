import React, { useState } from "react";
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";
import { IColumn } from "./interfaces";
import Column from "./Column";

interface BoardProps {
  columns: IColumn[];
}

const Board: React.FC<BoardProps> = ({ columns: initialColumns }) => {
  const [columns, setColumns] = useState(initialColumns);

  const moveCard = (
    cardId: string,
    sourceColumnId: string,
    targetColumnId: string
  ) => {
    setColumns((prevColumns) => {
      const sourceColumnIndex = prevColumns.findIndex(
        (col) => col.id === sourceColumnId
      );
      const targetColumnIndex = prevColumns.findIndex(
        (col) => col.id === targetColumnId
      );
      const sourceColumn = prevColumns[sourceColumnIndex];
      const targetColumn = prevColumns[targetColumnIndex];

      const cardIndex = sourceColumn.cards.findIndex(
        (card) => card.id === cardId
      );
      const card = sourceColumn.cards[cardIndex];

      const newSourceColumn = { ...sourceColumn };
      newSourceColumn.cards.splice(cardIndex, 1);

      const newTargetColumn = { ...targetColumn };
      newTargetColumn.cards.push(card);

      const newColumns = [...prevColumns];
      newColumns[sourceColumnIndex] = newSourceColumn;
      newColumns[targetColumnIndex] = newTargetColumn;

      return newColumns;
    });
  };

  return (
    <DndProvider backend={HTML5Backend}>
      <div style={{ display: "flex", gap: "16px" }}>
        {columns.map((column) => (
          <Column key={column.id} column={column} onCardDrop={moveCard} />
        ))}
      </div>
    </DndProvider>
  );
};

export default Board;
