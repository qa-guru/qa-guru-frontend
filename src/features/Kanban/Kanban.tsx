import React from "react";
import Board from "./Board";
import { IColumn } from "./interfaces";

const sampleData: IColumn[] = [
  {
    id: "1",
    title: "New",
    cards: [
      { id: "1", title: "Task 1", description: "This is task 1" },
      { id: "2", title: "Task 2", description: "This is task 2" },
    ],
  },
  {
    id: "2",
    title: "In review",
    cards: [{ id: "3", title: "Task 3", description: "This is task 3" }],
  },
  {
    id: "3",
    title: "Approved",
    cards: [{ id: "4", title: "Task 4", description: "This is task 4" }],
  },
  {
    id: "4",
    title: "Not approved",
    cards: [{ id: "5", title: "Task 5", description: "This is task 5" }],
  },
];

const Kanban = () => {
  return (
    <div style={{ padding: "16px" }}>
      <h1>Kanban Board</h1>
      <Board columns={sampleData} />
    </div>
  );
};

export default Kanban;
