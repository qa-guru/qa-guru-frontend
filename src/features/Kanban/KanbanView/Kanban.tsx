import React, { useState } from "react";
import { Grid, Stack, Typography } from "@mui/material";
import Board from "../KanbanContainer/Homeworks";
import TrainingSelection from "../KanbanContainer/TrainingsByMentor";
import LectureSelection from "../KanbanContainer/TrainingLectures";
import { SelectedTrainingIdContext } from "../../../context/SelectedTrainingIdContext";
import { SelectedLectureIdContext } from "../../../context/SelectedLectureIdContext";

const Kanban = () => {
  const [selectedTrainingId, setSelectedTrainingId] =
    useState<string | null>(null);
  const [selectedLectureId, setSelectedLectureId] =
    useState<string | null>(null);

  return (
    <SelectedTrainingIdContext.Provider
      value={{ selectedTrainingId, setSelectedTrainingId }}
    >
      <SelectedLectureIdContext.Provider
        value={{ selectedLectureId, setSelectedLectureId }}
      >
        <Typography variant="h4">Task Board</Typography>
        <Stack direction="row" spacing={3} mt="20px">
          <TrainingSelection />
          <LectureSelection />
        </Stack>
        <Board />
      </SelectedLectureIdContext.Provider>
    </SelectedTrainingIdContext.Provider>
  );
};

export default Kanban;
