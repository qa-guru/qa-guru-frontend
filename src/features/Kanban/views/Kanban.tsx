import React from "react";
import { Typography } from "@mui/material";
import Form from "./Form";
import Board from "../containers/Homeworks";
import { KanbanContext } from "../context/KanbanContext";
import useKanbanState from "../hooks/useKanbanState";

const Kanban = () => {
  const kanbanState = useKanbanState();

  return (
    <KanbanContext.Provider value={kanbanState}>
      <Typography variant="h4">Task Board</Typography>
      <Form />
      <Board />
    </KanbanContext.Provider>
  );
};

export default Kanban;
