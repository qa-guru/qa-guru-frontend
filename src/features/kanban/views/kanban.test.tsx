import { MemoryRouter } from "react-router-dom";
import Kanban from "./kanban";
import { render } from "../../../test/utils-test";

describe("Kanban", () => {
  it("kanban is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Kanban />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
