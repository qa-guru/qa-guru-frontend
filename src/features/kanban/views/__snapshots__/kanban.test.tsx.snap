// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kanban > renders correctly 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiStack-root css-1hhol0a-MuiStack-root"
    >
      <div
        aria-label="Выберите курс"
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        data-mui-internal-clone-element="true"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1uxbq4s-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          Выберите урок
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-disabled="true"
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled=""
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Выберите урок
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1uxbq4s-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          Выберите наставника
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-d4zdg1-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-1c0lsac-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Выберите наставника
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-hzmf2y-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1uxbq4s-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Созданные от
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-i81sip-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-7053f3-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            inputmode="text"
            placeholder="MM/DD/YYYY"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Созданные от
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-hzmf2y-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-1uxbq4s-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r3:"
          id=":r3:-label"
        >
          Созданные до
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-i81sip-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-7053f3-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r3:"
            inputmode="text"
            placeholder="MM/DD/YYYY"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="Choose date"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Созданные до
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-1pkdajw-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
  <div
    class="MuiGrid-root MuiGrid-container css-il5n0a-MuiGrid-root"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-d3oyhb-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
